pkg/flink/internal/controller/input_controller.go

```diff
@@ -190,7 +190,11 @@ func (c *InputController) getKeyBindings() []prompt.Option {
 			prompt.OptionAddKeyBind(prompt.KeyBind{
 				Key: prompt.ControlD,
 				Fn: func(b *prompt.Buffer) {
-					c.shouldExit = true
+					if b.Text() != "" {
+						b.Delete(1)
```

Comment:
lol, for some reason this is deleting two characters on native mac terminal

Reply:
https://github.com/confluentinc/cli/assets/11739405/caf7cb26-0f71-4a83-9c9e-f7d4f63154d7

Reply:
My first suspicion is that the keybind is somehow weirdly getting triggered twice

Reply:
Ohh. We don't need to have this keybinding at all since it's already covered by emacsKeyBindings. And somehow it still has the behaviour of ending the session when the prompt is empty (but I don't currently see the function that handles that)

Reply:
ah nvm found it - it's in prompt.go

Reply:
I don't know why it was working for me though.. but it seems to be working consistently now

Reply:
it does!